relay-server/src/metrics_extraction/sessions.rs

```diff
@@ -38,125 +196,163 @@ pub fn extract_session_metrics<T: SessionLike>(
         }
     };
 
-    let mut tags = BTreeMap::new();
-    tags.insert("release".to_owned(), attributes.release.clone());
-    if let Some(ref environment) = attributes.environment {
-        tags.insert("environment".to_owned(), environment.clone());
-    }
-    if let Some(client) = client {
-        tags.insert("sdk".to_owned(), client.to_owned());
-    }
+    let release = attributes.release.clone();
+    let environment = attributes.environment.clone();
+    let sdk = client.map(|s| s.to_owned());
```

Comment:
nit: these instances are cloned here to be later be either cloned again, or moved (`session[crashed]`). I would make them references here, and clone them everywhere below, i.e. `let release = attributes.release;`